src/QMCDrivers/WFOpt/QMCCostFunctionBase.cpp

```diff
@@ -88,19 +88,18 @@ QMCCostFunctionBase::~QMCCostFunctionBase()
     delete debug_stream;
 }
 
-void QMCCostFunctionBase::setRng(std::vector<RandomGenerator_t*>& r)
+void QMCCostFunctionBase::setRng(UPtrVector<RandomGenerator_t>& r)
 {
   if (MoverRng.size() < r.size())
   {
-    delete_iter(MoverRng.begin(), MoverRng.end());
-    delete_iter(RngSaved.begin(), RngSaved.end());
     MoverRng.resize(r.size());
     RngSaved.resize(r.size());
   }
   for (int ip = 0; ip < r.size(); ++ip)
-    MoverRng[ip] = r[ip];
+    MoverRng[ip] = std::move(r[ip]);
```

Comment:
It might work to make a new RNG and copy the state here (like RngSaved does below), rather than the transfer of ownership.
Alternately, make a new RNG and copy the state when RngCompatibility is populated in QMCDriverNew. (This might be better because splits Rng from RngCompatibility there).

These changes might alter test results if VMC get runs after an optimizer run, since the parent RNG won't have the state advanced during the optimizer.